go/src/koding/kites/kloud/provider/azure/stack.go

```diff
@@ -316,6 +333,17 @@ func (s *Stack) injectCloudInit(vm map[string]interface{}, name, kiteKeyName str
 	return countKeys, nil
 }
 
+func (s *Stack) id() string {
```

Comment:
Do you need it now? I think it would be better to just copy it.

It's good enough for Azure, but I don't find it a good API for generic purpose - the `(*BaseStack).ID()` does not say a thing, I'd rather add something like `func GenerateUniqueName(scope Scope) string` where Scope can be `StackScope`, `CredentialScope` etc. so one does not need to wonder if strings returned by `ID()` would conflict on reinit or not.